lib/dashboard.dart

```diff
@@ -41,6 +45,17 @@ class _DashboardState extends State<Dashboard> {
   void initState() {
     super.initState();
     _pageController = PageController();
+    try {
+      //As soon as the homescreen loads firstly all the user info is fetched from firbase
+      userdata.fetchData().then((value) {
```

Comment:
Instead of calling the method here, you can use the uid from StreamBuilder in main.dart and pass it to the fetch method.
Also, receive the user object and pass it down.